test/test_utils_magma.py

```diff
@@ -0,0 +1,391 @@
+            (CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS, ["test_0", "test_2", "test_1"]),
+        ],
+    )
+    def test_check_presence_of_attributes_no_errors(
+        self, input_object, attributes_to_check
+    ):
+        """
+        Test for function checking that specified attributes are part of a given object,
+        no errors raised.
+        """
+        result = check_presence_of_attributes(input_object, attributes_to_check)
+        assert result == None
+
+    @pytest.mark.parametrize(
+        "input_object, attributes_to_check",
+        [
+            (CLASSTESTER_OBJ_SINGLE_SIMPLE_ATTR, ["not_present"]),
+            (CLASSTESTER_OBJ_SINGLE_SIMPLE_ATTR, ["not_present_0", "not_present_1"]),
+            (CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS, ["test_0", "test_1", "not_present"]),
+            (
+                CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS,
+                ["test_0", "not_present", "test_1", "test_2"],
+            ),
+            (
+                CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS,
+                ["test_0", "not_present_0", "test_2", "not_present_1"],
+            ),
+            (
+                CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS,
+                ["not_present", "test_0", "test_2", "test_1"],
+            ),
+        ],
+    )
+    def test_check_presence_of_attributes_value_errors(
+        self, input_object, attributes_to_check
+    ):
+        """
+        Test for function checking that specified attributes are part of a given object,
+        ValueError raised.
+        """
+        with pytest.raises(ValueError) as value_err_info:
+            check_presence_of_attributes(input_object, attributes_to_check)
+        assert "Object validation error" in str(value_err_info.value)
+
+
+class TestSetListAttributes:
+    @pytest.mark.parametrize(
+        "input_object, attributes_to_set",
+        [
+            (CLASSTESTER_OBJ_SINGLE_SIMPLE_ATTR, None),
+            (CLASSTESTER_OBJ_SINGLE_SIMPLE_ATTR, []),
+            (CLASSTESTER_OBJ_SINGLE_SIMPLE_ATTR, ["test_0"]),
+            (CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS, ["test_0", "test_1"]),
+            (CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS, ["test_0", "test_1", "test_2"]),
+            (CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS, ["test_0", "test_2"]),
+            (CLASSTESTER_OBJ_SEVERAL_SIMPLE_ATTRS, ["test_0", "test_2", "test_1"]),
+        ],
+    )
+    def test_set_list_attributes_of_existing_nonlist_attributes(
+        self, input_object, attributes_to_set
+    ):
+        """
+        Test for function that gets rid of duplicates within object attributes that are lists,
+        or sets attributes to empty list if not present within the object.
+        Cases where the attributes to set are existent and are NOT lists, no action done.
+        """
+        original_object = deepcopy(input_object)
+        result = set_list_attributes(input_object, attributes_to_set)
+        assert result == None
```

Comment:
`result is None` preferred over `result == None` (latter will be caught by linters as not meeting PEP8 standards).